### PR TITLE
Feature/api main page

### PR DESCRIPTION
**Pull request by JJoeeyy**
### What?
We added the Player, with his Data
Also we add the Sections
the buttons are a pen and eraser now 👍 

### Why?
The Player is important, with the data from the BE we can now interact with the data

### How?
with a fetch to the BE

### Testing?
no

### Screenshots (for frontend changes)
no
 
### Anything Else?
Unfortuntely has the BackeEnd only two sections ready :(
Because the BackEnd had no CrossOrigin, we did it ourselfe :(

[//]: # (for a sample see: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/)

### CODE REVIEW

**Review by Lion-alt — APPROVED** [review]
LGTM 👍

File: fabbwled-frontend/src/routes/create/AdventureSheet.tsx [hunk 1]
```diff
@@ -44,7 +45,7 @@ export default function AdventureSheet() {
       const characterCreateDto: CharacterCreateDto = {
         player: {
           name: formik.values.name,
-          currentSection: { bookId: 0, sectionId: 0 },
+          currentSection: "1-1",
```
Comment (Noeell) [inline fabbwled-frontend/src/routes/create/AdventureSheet.tsx:48]: 🤝

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 2]
```diff
@@ -1,51 +1,139 @@
...
+      getSection(player.currentSection.split("-")[1]).then(setSection);
+    });
+  }, []);
+
+  const getCharacter = async () => {
+    const response = await fetch("http://localhost:8080/api/player");
+    return response.json();
+  };
+
+  console.log(character);
```
Comment (Noeell) [inline fabbwled-frontend/src/routes/game/Game.tsx:23]: Remove?

**Review by corsin-lehre — APPROVED** [review]
I think the 3 of us wrote pretty good code

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 3]
```diff
@@ -1,51 +1,139 @@
...
+    }).then((res) => res.json());
+  };
+
+  const getSection = async (sectionId: string): Promise<Section> => {
+    try {
+      const response = await fetch(
+        `http://localhost:8080/api/section/1/${sectionId}`,
+      ); //${character?.currentSection.sectionId}
+      const data = await response.json();
+      console.log(data);
```
Comment (Noeell) [inline fabbwled-frontend/src/routes/game/Game.tsx:56]: Remove?

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 4]
```diff
@@ -1,51 +1,139 @@
...
+                <ul>
+                  <li>Rank: {character?.rank}</li>
+                  <li>Profession: {character?.profession}</li>
+                  <li>Stamina: {character?.stamina}</li>
+                  <li>Defence: {character?.defence}</li>
+                  <li>Money: {character?.shards?.shardCount}</li>
+                </ul>
+                <ul>
+                  <li>Charisma: {character?.baseStats?.charisma}</li>
+                  <li>Combo: {character?.baseStats?.combat}</li>
```
Comment (Noeell) [inline fabbwled-frontend/src/routes/game/Game.tsx:106]:
```suggestion
                  <li>Comboat: {character?.baseStats?.combat}</li>
```

File: fabbwled-frontend/src/api/character.ts [hunk 5]
```diff
@@ -1,40 +1,33 @@
...
 
-    if (response.ok) {
-      return await response.json();
-    } else {
-      throw new Error(`API request failed with status: ${response.status}`);
-    }
-  } catch (error: any) {
-    // deliberately used any here
-    throw new Error(`An error occurred: ${error.message}`); // Handle network errors
+export async function setCharacter(data: CharacterCreateDto): Promise<unknown> {
```
Comment (Noeell) [inline fabbwled-frontend/src/api/character.ts:5]: How do you handle network errors now? (Without a try-catch)

**Review by Noeell — CHANGES_REQUESTED** [review]

File: fabbwled-frontend/src/api/character.ts (hunk 5 above)
Comment (Nilstrieb) [reply fabbwled-frontend/src/api/character.ts:5]: They're handled the same way as before, an error gets thrown. Just without the "an error occurred" message

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 6]
```diff
@@ -1,51 +1,139 @@
...
+      return data;
+    } catch (e) {
+      return {
+        body: {
+          type: "CONTAINER",
+          style: "NONE",
+          children: [
+            {
+              type: "SIMPLE",
+              text: "Haha existiert nicht",
```
Comment (Nilstrieb) [inline fabbwled-frontend/src/routes/game/Game.tsx:66]: This makes sense for a 404, but I don't think it makes sense for a network error, it should say something about network errors

**Review by Nilstrieb — COMMENTED** [review]
not sure why my review was requested

File: fabbwled-frontend/src/api/character.ts (hunk 5 above)
Comment (Noeell) [reply fabbwled-frontend/src/api/character.ts:5]: Alright 👍

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 7]
```diff
@@ -1,51 +1,139 @@
...
+  }, []);
+
+  const getCharacter = async () => {
+    const response = await fetch("http://localhost:8080/api/player");
+    return response.json();
+  };
+
+  console.log(character);
+  function rollDice(diceNumber: number) {
+    console.log(diceNumber);
```
Comment (oliver-jenny) [inline fabbwled-frontend/src/routes/game/Game.tsx:24]:
```suggestion

```

File: fabbwled-frontend/src/routes/game/Game.tsx (hunk 2 above)
Comment (oliver-jenny) [inline fabbwled-frontend/src/routes/game/Game.tsx]:
```suggestion

```

File: fabbwled-frontend/src/routes/game/Game.tsx (hunk 3 above)
Comment (Noeell) [reply fabbwled-frontend/src/routes/game/Game.tsx:56]:
```suggestion

```

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 8]
```diff
@@ -1,51 +1,139 @@
...
+    document.cookie = "";
+  }
+
+  function exitGame() {
+    document.cookie = "";
+    window.location.href = "about:blank";
+  }
+
+  const clickItem = (clickId: number) => {
+    console.log(clickId);
```
Comment (oliver-jenny) [inline fabbwled-frontend/src/routes/game/Game.tsx:37]:
```suggestion

```

File: fabbwled-frontend/src/routes/game/Game.tsx (hunk 3 above)
Comment (oliver-jenny) [inline fabbwled-frontend/src/routes/game/Game.tsx]:
```suggestion

```

File: fabbwled-frontend/src/routes/game/Game.tsx (hunk 2 above)
Comment (Noeell) [reply fabbwled-frontend/src/routes/game/Game.tsx:23]:
```suggestion

```

**Review by Noeell — CHANGES_REQUESTED** [review]

**Review by Noeell — APPROVED** [review]

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 9]
```diff
@@ -1,51 +1,139 @@
...
+    console.log(diceNumber);
+  }
+
+  function newGame() {
+    document.cookie = "";
+  }
+
+  function exitGame() {
+    document.cookie = "";
+    window.location.href = "about:blank";
```
Comment (oliver-jenny) [inline fabbwled-frontend/src/routes/game/Game.tsx:33]: With window.close() you should be able to close the Tab -> is more sensible imo

```suggestion
   window.close();
```

**Review by oliver-jenny — APPROVED** [review]
Lgtm - still look at comments -> nothing big, therefore still aproved

File: fabbwled-frontend/src/routes/game/Game.tsx (hunk 9 above)
Comment (corsin-lehre) [reply fabbwled-frontend/src/routes/game/Game.tsx:33]: https://stackoverflow.com/questions/19761241/window-close-and-self-close-do-not-close-the-window-in-chrome

Comment (oliver-jenny) [reply fabbwled-frontend/src/routes/game/Game.tsx:32]: Aight then its ok

File: fabbwled-frontend/src/routes/game/Game.tsx [hunk 10]
```diff
@@ -1,51 +1,135 @@
...
+      getSection(player.currentSection.split("-")[1]).then(setSection);
+    });
+  }, []);
+
+  const getCharacter = async () => {
+    const response = await fetch("http://localhost:8080/api/player");
+    return response.json();
+  };
+
+  function rollDice(diceNumber: number) {
```
Comment (Noeell) [inline fabbwled-frontend/src/routes/game/Game.tsx:23]: Function is empty, that’s why the Action fails 🚨

**Review by nnikolajj — APPROVED** [review]

**Pull request merged**